### PR TITLE
WT-10523 Fail s_clang_format gracefully on unsupported arch

### PR DESCRIPTION
**Pull request by etienneptl**
The changes:
- Improve the error message when s_clang_format.sh cannot be executed on the platform
- Add exception handling
- Rewrite the s_clang_format.sh to hopefully increase readability

### CODE REVIEW

**Review by keitharnoldsmith — APPROVED** [review]
Thanks Etienne!

LGTM

File: dist/s_clang_format [hunk 1]
```diff
@@ -29,23 +42,16 @@ if [ -n "$is_mongo_repo" ] ; then
...
 cd "$dest_dir" || exit 1
 
-# Override existing Clang Format versions in the PATH.
-export PATH="${PWD}/dist":$PATH
-
-# Download the clang-format binary if it's not in place.
-[ ! -x "$(command -v clang-format)" ] && download_clang_format
-
-# Ensure that we have the correct version of clang-format.
+# Check if Clang-Format is already available with the desire version.
```
Comment (lukech) [inline dist/s_clang_format:45]: desire->desired

File: dist/s_clang_format [hunk 2]
```diff
@@ -6,16 +6,29 @@ t=__wt.$$
...
-        tar --strip=1 -C dist/ -xf dist/clang-format.tar.gz clang-format-llvm-10.0.0-x86_64-apple-darwin/clang-format && rm dist/clang-format.tar.gz
+    version=$1
+    # FIXME-WT-10583
+    arch=$(uname -m)
+    if [ ! "$arch" = "x86_64" ]; then
+        echo "$0: unsupported architecture $arch"
+        return 1
+    fi
+
+    archive=./dist/clang-format.tar.gz
```
Comment (lukech) [inline dist/s_clang_format:17]: The below curl & tar commands use the variable with `./` preceded. Is the `./` here in variable value necessary?

File: dist/s_clang_format [hunk 3]
```diff
@@ -29,23 +42,16 @@ if [ -n "$is_mongo_repo" ] ; then
...
-# Ensure that we have the correct version of clang-format.
+# Check if Clang-Format is already available with the desire version.
 desired_version="10.0.0"
+if ! command -v clang-format &> /dev/null; then
+    download_clang_format $desired_version || exit 1
+elif ! clang-format --version | grep -q $desired_version; then
+    download_clang_format $desired_version || exit 1
+fi
 
-# On macOS Catalina, users need to manually approve binaries.
```
Comment (lukech) [inline dist/s_clang_format:41]: It feels this comment is still useful, for folks who start building wiredtiger on macOS.

**Review by lukech — APPROVED** [review]
LGTM modulo a few nits. Thanks.

File: dist/s_clang_format (hunk 1 above)
Comment (etienneptl) [reply dist/s_clang_format]: Thanks! 5af36d4764b48c0291ca3bdf4fb8926dd2065184

File: dist/s_clang_format (hunk 2 above)
Comment (etienneptl) [reply dist/s_clang_format]: It's not, update here 5af36d4764b48c0291ca3bdf4fb8926dd2065184

File: dist/s_clang_format (hunk 3 above)
Comment (etienneptl) [reply dist/s_clang_format:41]: I added it back being more general, see 5af36d4764b48c0291ca3bdf4fb8926dd2065184

**Pull request merged**